### PR TITLE
Cleanup / move some serverless code to ES

### PR DESCRIPTION
**Pull request by mosche**
Moving this to prepare some minor cleanup of the metering code

### CODE REVIEW

**Review by ldematte — COMMENTED** [review]
Thanks for doing this!
Can you move the tests in `IteratorsTests` too?

**Comment by mosche** [comment]
Thanks for noticing @ldematte 🤦 
Also quick note, i noticed composition was kind of broken. I've removed it, but let me know if you think it should be fixed instead:
Assume a limit iterator of limit n, consume 1 element, create another limit iterator of limit n of that one. You'll be able to consume another n instead of n-1 elements...

**Comment by mosche** [comment]
@elasticmachine update branch

**Comment by ldematte** [comment]
> I've removed it, but let me know if you think it should be fixed instead

Idk, for our use case is fine since we do not use anything similar. I see that other iterators do handle composition (like map), but I think it's much more likely to have map(map()) than limit(limit()).
TL;DR: I think it's OK to remove it, I'll add composition back if/when I have time (and take the partial consumption case into consideration)

**Review by ldematte — APPROVED** [review]
